configure.ac

```diff
@@ -4849,6 +4849,55 @@ if test "$ac_cv_func_thread_policy_set" = yes ; then
     fi
 fi
 
+AC_ARG_ENABLE(internal-openmp,
+    [--enable-internal-openmp
+       Enable MPICH to be OpenMP aware
+       level:
+         yes       - Enabled (includes -fopenmp flag)
+         no        - Disabled (default)
+    ],,enable_internal_openmp=no)
+
+if test "$enable_internal_openmp" = "yes"; then
+    AC_DEFINE([MPIDI_ENABLE_INTERNAL_OPENMP], [1], [Define when MPICH is OpenMP aware])
+
+    mpidi_check_compiler_vendor="unknown"
+
+    # Intel
```

Comment:
It would be preferable to pull in `AX_OPENMP` to determine the right flags for this PR. https://www.gnu.org/software/autoconf-archive/ax_openmp.html